Improve some explanations in README

higucheese opened the pull request:
Changes in this pull request
 - Add a short explanation for `m_log_data_t` type
 - Make some explanations of "Parameters" more specific
 - Fix an expression ("A pointer" → "Pointer")

higucheese committed:
Change explanations in README

s417-lama commented on README.md:
```diff
@@ -34,36 +34,38 @@ make check
-* `buf_size`  : Size of the begin buffer and the end buffer.
+* `md`        : Pointer to a `mlog_data_t` variable at global scope.
+* `num_ranks` : The number of ranks (e.g., the number of workers/threads using MassiveLogger).
+* `buf_size`  : Initial size of buffering memories in the `mlog_data_t` variable.
```
The plural form "memories" sounds a bit strange for me. Maybe it is uncountable in computing.

By the way, we should have noted that it allocates
- two buffers that have `size` bytes of memory respectively
- per rank,
- so the total size of memory will be 2 * `size` * `num_ranks`.

higucheese replied:
> The plural form "memories" sounds a bit strange for me. Maybe it is uncountable in computing.

That is certainly true. The "memories" word is not proper for explaining this kind of thing. I will change the expression of that.

I also agree to note what the function allocates internally.
However, using "begin buffer" and "end buffer" words at the beginning of README can confuse readers since they are not explained before.
I should take care of word choice for a reader-friendly explanation.

s417-lama replied:
You are absolutely right. The problem here is that we don't explain how it works before API specs, which should be an another issue.

s417-lama reviewed (changes requested):
Thank you for your contribution!!
Good documentation is really important. If you encounter any unclear thing, please add more information!

higucheese committed:
Update README

higucheese commented:
I changed not to use the plural form and added more detailed information. I avoid using the "begin/end buffer" words on purpose because there is no explanation about them before.

Would you review this change?

s417-lama commented:
Great! Thank you so much!
